Modelica/Fluid/Machines.mo

```diff
@@ -311,6 +311,8 @@ Then the model can be replaced with a Pump with rotational shaft or with a Presc
     parameter Medium.MassFlowRate m_flow_start = system.m_flow_start
         "Guess value of m_flow = port_a.m_flow"
       annotation(Dialog(tab = "Initialization"));
+    parameter Types.CheckValveHomotopyType checkValveHomotopy = Types.CheckValveHomotopyType.NoHomotopy "= whether the valve is Closed, Open, or unknown at initialization"
+      annotation(Dialog(tab = "Initialization"));
```

Comment:
Adding new components to (base) classes is - strictly seen - not legal for maintenance releases.

Reply:
@beutlich, I know. I'm inclined to make an exception here, since 3.2.3 is an end-of-line maintenance versions which will have no backwards compatible new releases, i.e. 3.3.0, where I can put in this change. If we want to keep 3.2.3 in the long term and keep fixing it, we have to make some compromise.

Adding homotopy here is essential to get robust intialization into 3.2.3 maintenance - the extra parameter is essential because there are different homotopies, depending on the expected initial condition, and I need to select one.

The default value corresponds to the old behaviour, so the change is 100% backwards compatible.

If you see any other solution to fix this issue, I'm fine with it. Otherwise I think it's better to have a more robust MSL 3.2.3 than strictly respecting formal rules in a borderline case as MSL 3.2.3 maintenance.

What do you think?

Reply:
It is a trade-off which I believe @MartinOtter should decide in the end.

Reply:
I think in this very special situation it is fine to make an exception to the formal rules and I agree with Francescos proposal.

Reply:
Usually we add such backward-compatibility breaking changes in the release notes.

Reply:
@beutlich, I added a new parameter to an existing model, with a default value that keeps the existing behaviour unchanged. I'm not sure I understand how this could break backwards-compatibility. Could you please elaborate?

Reply:
Please see the appropriate [documentation](https://doc.modelica.org/Modelica%204.0.0/Resources/helpOM/Modelica.UsersGuide.ReleaseNotes.VersionManagement.html) (added by bf8b927c82c024c2f3c1ae2261fa4dad84d343ec for MSL v3.1).

> As a recommendation, a valid bug-fix to the maintenance branch may contain one or more of the following changes.
> * [..]
> * Introducing a new name in the public section of a class (model, package, ...) or in any section of a partial class is **not** allowed. Since otherwise, a user might use this new name and when storing its model and loading it with an older bug-fix version, an error would occur.

Reply:
I think the reference to _class_ is unnecessarily too generic.

I agree this is the case if we add a new _model_ to a _package_ class. But I really can't see how this could be a problem if we add a new _parameter_ to a _model_ class. How could a user use this parameter name? The only possible cause of conflict could be if the user extended the model and added a new parameter to it, which happened to have the same name, but a different meaning. Of course we can't rule this out 100%, but I would say this is really unlikely.

Do I miss something? Should we perhaps update that documentation?

Reply:
The main reason is, that models saved in MSL v3.2.3 Build 5 will no longer be compatible with MSL v3.2.3 Build 4. Usually we claim this kind of compatibility.

Reply:
Aha. I always thought that being backwards compatible meant that I could use the new library with an old model, not the other way round.

But then this would also apply to minor releases. Aren't we claiming that they are backwards compatible, according to semver rules?

Reply:
The expectation is/was, that no new components are introduced between different builds of MSL v3.2.3 (or according to semver, between different patch releases of MSL v4.0.x).